packages/qdrant-loader/src/qdrant_loader/core/state/checkpoint_manager.py

```
@@ -0,0 +1,286 @@
+
+            await self.session.commit()
+        except Exception as e:
+            await self.session.rollback()
+            logger.error(
+                "Failed to save checkpoint",
+                project_id=checkpoint.project_id,
+                source_type=checkpoint.source_type,
+                source=checkpoint.source,
+                error=str(e),
+                error_type=type(e).__name__,
+            )
```

Ruff (0.15.20)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)